docs/fixing-warnings.md

```diff
@@ -0,0 +1,189 @@
+
+# Trim warnings in .NET 6
+
+In .NET Core 3.1 and 5.0 we introduced trimming as a new preview feature for self-contained .NET
+core applications. Conceptually the feature is very simple: when publishing the application the
+.NET SDK analyzes the entire application, including the .NET framework, and removes all unused
+code. In the time trimming has been in preview, we've learned that trimming is very powerful --
+it can reduce application size by half or more. However, we've also learned about the
+difficulties in safely trimming applications.
+
+The most difficult question in trimming is what is unused, or more precisely, what is used. For
+most standard C# code this is trivial -- the trimmer can easily walk method calls, field and
```

Comment:
What is "the trimmer" ? We have not been using that term and most developers are familiar with linker term. Do we really have to confuse the community with another new name?

Reply:
In all locations other than `mono/linker` we've been trying to avoid the term "linker" because most people understand it as the "platform linker". We use "trim tool" or sometimes "trimmer" (I would prefer "trim tool" if possible).
For the same reason we use "trimming" and not "linking" everywhere.

Reply:
> we've been trying to avoid the term "linker" because most people understand it as the "platform linker"

Are you sure? Few references to public documentation that clearly uses IL Linker term to describe the tool.

- https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/configure-linker?view=aspnetcore-3.1
- https://www.telerik.com/blogs/reduce-self-contained-app-size-with-latest-net-core-features
- https://www.meziantou.net/optimizing-a-blazor-webassembly-application-size.htm
- https://montemagno.com/optimizing-xamarin-apps-libraries-with-the-linker/
- https://docs.telerik.com/devtools/xamarin/knowledge-base/common-understanding-ui-for-xamarin-and-linker-behavior

Reply:
I'm intentionally "ignoring" the mono/Xamarin side of things - those will likely use "linker" a lot. Most of our docs and blog posts use "Trimming" - we also use it everywhere in the messages (RUC) in the framework. I don't see us going back to "linking".

That said - I understand the confusion for Xamarin users - we might want to mention this on the first use of the term.

Reply:
I was not referring to trimming but to "trimmer". Even .NET Core MSDN documentation is written in the way that it uses the term since 3.1.

> .NET Core 3+ ships with an IL Linker which allows you to reduce the size of the self-contained application. It automatically determines which assemblies are needed and copies only them.

I'm really not sure we have to rename everything to cause more confusion.

Reply:
I would probably just not give it a name here. We can update the sentence to say "For most standard C# code this is trivial -- ~~the trimmer~~we can easily walk method calls, field and"

That said,

https://en.wikipedia.org/wiki/Linker_(computing)

This is the definition of the term used in computer engineering. It's not what IL Linker is doing. It's confusing to anyone who has ever used a linker in the past.

Reusing this term in .NET was a mistake. We should have gone with something like [Remover 8000](https://github.com/mono/embeddinator-4000). Now Xamarin often needs to qualify the word "linker" with the word "native" when referring to what everyone outside .NET calls "linker". It's a pretty sad place to be in and puts us (the authors of .NET) into a bad light (trying to steal a well-established term).

I've seen this cause confusion in the past. "Linker failed. Oh, not that one, the other one."

We should phase this name out before it moves out of the niche parts of .NET (both Xamarin and Blazor-WASM are still niches).

Reply:
My use of the term "trimmer" here is in its most generic sense -- it is the thing which "trims." The other documentation about a specific tool which is responsible for trimming is incorrect in this case because the only supported method of trimming applications in .NET 6 will be passing `PublishTrimmed=true` to the SDK. Rather than talk specifically about the SDK, I wanted to just use a generic term.

I don't really see the need to get more specific here and talk about any specific tool. The term "linker" is certainly incorrect for .NET 6 since we do not plan on shipping a separate linker tool for supported scenarios.